addons/binding/org.openhab.binding.xiaomivacuum/src/main/java/org/openhab/binding/xiaomivacuum/internal/Utils.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.xiaomivacuum.internal;
+
+import java.util.Date;
+
+/**
+ * Utility class for common tasks within the MAX! binding package.
+ *
+ * @author Andreas Heil (info@aheil.de)
+ * @author Marcel Verpaalen
+ *
+ * @since 1.4.0
+ */
+
+public final class Utils {
+
+    /**
+     * Returns the integer value of an hexadecimal number (base 16).
+     *
+     * @param hex
+     *            the hex value to be parsed into an integer
+     * @return the given hex value as integer
+     */
+    public static final int fromHex(String hex) {
+        return Integer.parseInt(hex, 16);
+    }
+
+    /**
+     * Returns the hexadecimal number of a number of integer values.
+     *
+     * @param values
+     *            the integer values to be converted into hexadecimal numbers
+     * @return the given numbers as hexadecimal number
+     */
+    public static final String toHex(int... values) {
+        String returnValue = "";
+        for (int v : values) {
+            returnValue += v < 16 ? "0" + Integer.toHexString(v).toUpperCase() : Integer.toHexString(v).toUpperCase();
+        }
+        return returnValue;
+    }
+
+    /**
+     * Returns the hexadecimal number of a bit array .
+     *
+     * @param bits
+     *            the boolean array representing the bits to be converted into
+     *            hexadecimal numbers
+     * @return the hexadecimal number
+     */
+    public static final String toHex(boolean[] bits) {
+        int retVal = 0;
+        for (int i = 0; i < bits.length; ++i) {
+            retVal |= (bits[i] ? 1 : 0) << i;
+        }
+        return toHex(retVal);
+    }
+
+    /**
+     * Converts an Java signed byte into its general (unsigned) value as being
+     * used in other programming languages and platforms.
+     *
+     * @param b
+     *            the byte to be converted into its integer value
+     * @return the integer value represented by the given byte
+     */
+    public static final int fromByte(byte b) {
+        return b & 0xFF;
+    }
+
+    /**
+     * Resolves the date and time based based on a three byte encoded within a
+     * MAX! Cube L message.
```

Comment:
yes, it is... I now took out all of the ones that are not used in this binding and changed the text